# Pricing non linéaire avec le solveur Pyomo
# "Pricing " non linéaire d'un seul objet

Dnas l'exercice avec le solver GEKKO, on ne trouve pas pareil que dans le résultat Excel, qui lui arrive à mieux optimiser.
essayons de voir si Pyomo se rapporche plus du résultat du solveur Excel.

In [32]:
import pyomo.environ as pyo
import sys


# On configure le chemin du solveur non linéaire sous windows.
solvername='ipopt'
solverpath_folder='C:\\ipopt' #does not need to be directly on c drive
solverpath_exe='C:\\ipopt\\bin\\ipopt' #does not need to be directly on c drive
sys.path.append(solverpath_folder)


# On crée un modèle Pyomo
model = pyo.ConcreteModel()

# On crée une variable simple Pyomo
model.item = pyo.Var(within=pyo.PositiveReals, bounds=(0, 100000), initialize=1)

# On crée notre objectif
model.OBJ = pyo.Objective(expr = ((model.item - 50) * (1100 - 10 * model.item)), sense=pyo.maximize)


# On résouds
results = pyo.SolverFactory(solvername,executable=solverpath_exe).solve(model)

# On affiche les résultats.
print("item=", pyo.value(model.item))
print("objective=", pyo.value(model.OBJ))
print(results)
# model.pprint()




item= 80.00000000002473
objective= 9000.0

Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 0
  Number of variables: 1
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.11.1\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.10571432113647461
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



Il trouve pareil que Gekko !
item= 80.00000000002473
objective= 9000.0

ALors que Excel trouve un meilleur résultat!